### PR TITLE
Fix NATS/MeshSync connectivity test

### LINKED ISSUE #5836: NATS ping event should check the healthy state of NATS network connection and pod status
The NATS ping button should ping and check the health status of connection to NATS server and as a result the Pod status and remains. 
Screenshot 2022-07-18 at 9 44 53 PM

Even on deleting the pods, Meshsync still pingable
#### Environment

- **Host OS:** Mac Linux Windows
- **Platform:** Docker or Kubernetes
- **Meshery Server Version:** stable-v
- **Meshery Client Version:** stable-v

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by MUzairS15**
Signed-off-by: MUzairS15 <muzair.shaikh810@gmail.com>

**Description**

This PR fixes #5836 

1. Get the external IP and get the response of `<external-IP:8222/connz>`
2. If MeshSync is sending data, the endpoint will have MeshSync as client listed.
3. If Meshery Server is connected to Broker and receiving data, the endpoint will have meshery-server listed.

**Notes for Reviewers**
Dependent PR: https://github.com/meshery/meshkit/pull/197
**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
MeshKit should be used here.

File: internal/graphql/model/operator_helper.go [hunk 1]
```diff
@@ -103,11 +114,14 @@ func GetBrokerInfo(mesheryclient operatorClient.Interface, mesheryKubeClient *me
 	brokerVersion := ""
 	if err == nil {
 		brokerVersion = imageVersionExtractUtil(statefulSet.Spec.Template, "nats")
-	}
-	if err == nil {
-		status := fmt.Sprintf("%s %s", StatusConnected, broker.Status.Endpoint.External)
-		if brokerConn.Info() != brokerpkg.NotConnected {
-			brokerStatus.Status = Status(status)
+		externalIP := strings.Split(broker.Status.Endpoint.External, ":")[0]
+		endpoint := fmt.Sprintf("http://%s%s/%s", externalIP, ":8222", "connz")
```
Comment (leecalcote) [inline internal/graphql/model/operator_helper.go:118]: Too fragile.

**Comment by leecalcote** [comment]
@sudo-NithishKarthik please review this one.

**Comment by leecalcote** [comment]
> MeshKit should be used here.

@piyushsingariya I wonder if you'd like to offer any suggestions here.

**Comment by sudo-NithishKarthik** [comment]
> MeshKit should be used here.

I agree

File: internal/graphql/model/operator_helper.go [hunk 2]
```diff
@@ -103,11 +114,14 @@ func GetBrokerInfo(mesheryclient operatorClient.Interface, mesheryKubeClient *me
...
+
+		if connectivityTest(models.MesheryServerBrokerConnection, endpoint) {
```
Comment (sudo-NithishKarthik) [inline internal/graphql/model/operator_helper.go:120]: Looks like we only have logic for `StatusConnected` and/or its negation. But we also have `Deployed`, `NotDeployed` etc., right?

**Review by leecalcote — DISMISSED** [review]
Thank you, @Chadha93. @MUzairS15, I'll leave it to you to merge...

**Comment by MUzairS15** [comment]
@leecalcote ,Before merging meshkit release is required. I will pull @Revolyssup to finish review for meshkit PR, and then will merge this.

**Comment by MUzairS15** [comment]
@Revolyssup Please review before some conflicts appear,

**Comment by leecalcote** [comment]
@Revolyssup you missed this before another conflict arose. 

Please review.

**Comment by CyrineG** [comment]
@MUzairS15 after I run minikube tunnel, I still get a "`Meshery Broker could not be reached`" error and broker's status is stuck at `deployed ` in the UI, even though `<external-IP:8222/connz>` is accessible using curl and in the operator and meshery-server logs they are connecting to the broker :thinking:

**Comment by leecalcote** [comment]
@CyrineG ^^

**Comment by CyrineG** [comment]
@MUzairS15 

yes, in the logs I have :
```
INFO[2022-10-06T08:22:38Z] Connected to broker at:192.168.49.2:30606     app=meshery
INFO[2022-10-06T08:22:38Z] Connected broker endpoints : 192.168.49.2:30606  app=meshery
```

and in the `/connz` reply meshery-server is listed as a client
```
    {
      "cid": 9,
      "kind": "Client",
      "type": "nats",
      "ip": "172.17.0.1",
      "port": 51096,
      "start": "2022-10-06T08:22:42.832589104Z",
      "last_activity": "2022-10-06T08:22:42.8336968Z",
      "rtt": "328µs",
      "uptime": "1m4s",
      "idle": "1m4s",
      "pending_bytes": 0,
      "in_msgs": 1,
      "out_msgs": 0,
      "in_bytes": 136,
      "out_bytes": 0,
      "subscriptions": 2,
      "name": "meshery-server",
      "lang": "go",
      "version": "1.14.0"
    },
```

**Comment by leecalcote** [comment]
@Abhishek-kumar09 @MUzairS15 where do we take this from here given @Abhishek-kumar09 recent changes?

**Comment by leecalcote** [comment]
@Abhishek-kumar09 Mismatched states between the switcher and the settings page.

Screen Shot 2022-10-09 at 8 30 45 AM


@CyrineG are you going to document these statuses?

**Comment by CyrineG** [comment]
@leecalcote yes I'm working on a document that will have all currently existing states in UI and backend

**Comment by MUzairS15** [comment]
@leecalcote Have you tested this, I think we are good here (after resolving conflicts).

**Comment by leecalcote** [comment]
> @leecalcote Have you tested this, I think we are good here (after resolving conflicts).

Wonderful, @MUzairS15

**Comment by Revolyssup** [comment]
@MUzairS15 The response corresponds to what's shown in UI for MeshSync ping which is `deployed` but on ping the snackbar says that "MeshSync is not publishing data" although it is.
Screenshot 2023-02-21 at 4 11 50 PM
Screenshot 2023-02-21 at 4 11 32 PM

**Comment by MUzairS15** [comment]
// @Revolyssup

**Review by Revolyssup — APPROVED** [review]

**Pull request merged**